### PR TITLE
feat: migrate git coin to new plugin infra

### PR DESCRIPTION
**Pull request by UncleBill**
part of #3066

### CODE REVIEW

File: packages/maskbook/src/plugins/Gitcoin/base.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+export const base: Plugin.Shared.Definition = {
+    ID: PLUGIN_ID,
+    icon: PLUGIN_ICON,
+    name: { fallback: PLUGIN_NAME },
+    description: { fallback: PLUGIN_DESCRIPTION },
+    publisher: { name: { fallback: 'Mask Network' }, link: 'https://mask.io/' },
+    enableRequirement: {
+        architecture: { app: false, web: true },
+        networks: { type: 'opt-out', networks: {} },
+        target: 'insider',
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/Gitcoin/base.ts:13]:
```suggestion
        target: 'stable',
```

Comment (UncleBill) [reply packages/maskbook/src/plugins/Gitcoin/base.ts]: Fixed in https://github.com/DimensionDev/Maskbook/pull/3495/commits/804dac51f21d5506d29f8f214d703420b6b3816a

File: packages/maskbook/src/plugins/Gitcoin/SNSAdaptor/index.tsx [hunk 2]
```diff
@@ -1,45 +1,39 @@
...
             .find(isGitcoin)
         if (!link) return null
         return <Renderer url={link} />
     },
-    PageComponent() {
-        return <DonateDialog />
-    },
-    DashboardComponent() {
-        return <DonateDialog />
-    },
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/Gitcoin/SNSAdaptor/index.tsx:42]: Still need to add a Dashboard entry to the new plugin.

Comment (UncleBill) [reply packages/maskbook/src/plugins/Gitcoin/SNSAdaptor/index.tsx:42]: I think it's not required for Gitcoin, we don't have any Gitcoin dialog in dashboard.

Comment (Jack-Works) [reply packages/maskbook/src/plugins/Gitcoin/SNSAdaptor/index.tsx:42]: 🤔So why is that added in the first time? Do you know @guanbinrui ?

Comment (guanbinrui) [reply packages/maskbook/src/plugins/Gitcoin/SNSAdaptor/index.tsx:42]: Agree with @UncleBill. I may add it arbitrarily.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**